modules/public/public.go

```diff
@@ -98,76 +57,51 @@ func parseAcceptEncoding(val string) map[string]bool {
 	return types
 }
 
-func (opts *Options) handle(w http.ResponseWriter, req *http.Request, opt *Options) bool {
+func (opts *Options) handle(w http.ResponseWriter, req *http.Request, fs http.FileSystem, prefix string) bool {
 	if req.Method != "GET" && req.Method != "HEAD" {
 		return false
 	}
 
 	file := req.URL.Path
 	// if we have a prefix, filter requests by stripping the prefix
-	if opt.Prefix != "" {
-		if !strings.HasPrefix(file, opt.Prefix) {
+	if prefix != "" {
+		if !strings.HasPrefix(file, prefix) {
 			return false
 		}
-		file = file[len(opt.Prefix):]
+		file = file[len(prefix):]
 		if file != "" && file[0] != '/' {
 			return false
 		}
 	}
 
-	f, err := opt.FileSystem.Open(file)
+	f, err := fs.Open(file)
 	if err != nil {
-		// 404 requests to any known entries in `public`
-		if path.Base(opts.Directory) == "public" {
-			parts := strings.Split(file, "/")
-			if len(parts) < 2 {
-				return false
-			}
-			for _, entry := range KnownPublicEntries {
-				if entry == parts[1] {
-					w.WriteHeader(404)
-					return true
-				}
-			}
+		if os.IsNotExist(err) {
```

Comment:
We should consider handling:

```go
err.Error() == "http: invalid character in file path"
```

on Windows.

Or at line 71 do a `filepath.ToSlash` to avoid having this error ever occur.

Reply:
I think we should not accept requests like `/assets\img\404.png`. So `filepath.ToSlash` should not be there.

Reply:
that's fine but we should probably StatusNotFound instead of StatusInternalServerError in this case.

Reply:
Hm, maybe `StatusBadRequest` is better.

Reply:
Check added on line 56